mitodl/teachersportal
# Implemented registration

#### What's this PR do?
- Adds REST APIs for registration and activation
- Updates the UI to provide a registration form and to accept activation tokens
- Sends the registration email
#### Where should the reviewer start?

`portal/views.py` and `static/js/containers/ActivatePage.js`
#### How should this be manually tested?
- Start with incorrect email settings in order to make sure registration fails in a correct manner.
- Log in to Django shell and verify that no user exists with the given email address.
- Go to the course detail page for any course. Click on 'Sign in or register' in the header.
- Fill in the registration fields using your MIT email as the email address. Click 'Register' and verify that an error message shows up. In the Django shell verify that no new user was created.
- Set up your email settings for teacher's portal. One option is to allow access to a gmail account, or you can set up a local SMTP server.
- Fill in the registration fields again and verify that the login dialog disappears and a registration success message shows up in the body of the page.
- Go to `/oscar/` and attempt to login with your credentials. It should say the account is inactive.
- Verify that you received an email that looks like a registration email and that it contains a link. Click the link. Verify that a success message appears and that you are redirected to the old page after a few seconds.
- Attempt to login and verify that this succeeds.
#### Any background context you want to provide?

We are using email addresses as usernames. For the Django User table this limits it to certain characters and up to 30 characters in length. We may want to replace the default User table if these limits are unacceptable. (#141)
#### What are the relevant tickets?

Fixes #121
